pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -647,9 +645,11 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 
 // PublishAppInstMetaDataToZedCloud is called when an appInst reports its Metadata to EVE.
 // AppInst metadata is relayed to the controller to be processed further.
-func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstID string, appInstMetadata *types.AppInstMetaData,
-	metadataType types.AppInstMetaDataType, iteration int) {
-	log.Functionf("PublishAppInstMetaDataToZedCloud: appInstID: %v", appInstID)
+func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstMetadata *types.AppInstMetaData, isDelete bool) {
+
+	metadataType := appInstMetadata.Type
+	appInstId := appInstMetadata.AppInstUUID.String()
```

Comment:
FWIW this would currently core dump due to a nil pointer in the delete case. That is why we do need the is_delete boolean.

Reply:
oops, sorry I accidentally lost that piece of code. It never supposed to be a nil